clusterloader2/pkg/framework/client/objects.go

```diff
@@ -172,7 +172,10 @@ func PatchObject(dynamicClient dynamic.Interface, namespace string, name string,
 func DeleteObject(dynamicClient dynamic.Interface, gvk schema.GroupVersionKind, namespace string, name string) error {
 	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
 	deleteFunc := func() error {
-		return dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		// Delete operation removes object with all of the dependants.
+		falseVar := false
+		deleteOption := &metav1.DeleteOptions{OrphanDependents: &falseVar}
+		return dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, deleteOption)
```

Comment:
Can you split all of the above to a separate PR?

I think this is much less controversial.
Wrt to changes to the test itself, I think it requires significantly more work.

Reply:
Code changes moved to #255.